libr/bin/format/elf/elf.c

```diff
@@ -1507,177 +1486,70 @@ static ut64 get_import_addr_x86(ELFOBJ *bin, struct ht_rel_t *rel, RBinElfSectio
-				if (len > 0 && buf[0] != 0xff) {
-					// the .plt section has been probably split in
-					// .plt and .plt.sec to support Intel MPX. See
-					// https://github.com/hjl-tools/x86-psABI
-					//
-					// This is the new layout of the .plt
-					//
-					//     endbr64
-					//     push 0
-					//     bnd jmp 0x4020
-					//
-					// The .plt.sec section has something like:
-					//
-					//     endbr64
-					//     bnd jmp qword reloc.__ctype_toupper_loc
-					//
-					if (pltsec_section) {
-						// if possible, returns the address in
-						// .plt.sec, which is the address called
-						// by call instructions
-						return pltsec_section->rva + rel->k * 16;
-					}
-					// plt_sym_addr in this case points to the start
-					// of .plt entry
-					return plt_sym_addr;
-				}
-			}
-			// this is a regular .plt section
-			//
-			//     jmp qword reloc.__ctype_toupper_loc
-			//     push 0
-			//     jmp 0x4020
-			//
-			return plt_sym_addr - 6;
-		}
-	default:
-		bprintf ("Unsupported relocation type for imports %d\n", REL_TYPE);
-		return of;
-	}
-}
+		size_t i = 0;
+		plt_sym_addr = READWORD (buf, i);
 
-static ut64 get_import_addr_riscv(ELFOBJ *bin, struct ht_rel_t *rel, RBinElfSection *plt_section) {
-	RBinElfSection *s = get_section_by_name (bin, ".rela.plt");
-	if (s) {
-		ut8 buf[1024];
-		const ut8 *base;
-		ut64 plt_addr = s->rva + s->size;
-		int len = r_buf_read_at (bin->b, s->offset + s->size, buf, sizeof (buf));
-		if (len != sizeof (buf)) {
-			// oops
+		//relative address
+		if ((plt_addr + 6 + Elf_(r_bin_elf_v2p) (bin, plt_sym_addr)) == rel->rva) {
+			return plt_addr;
 		}
-		base = r_mem_mem_aligned (buf, sizeof (buf), (const ut8 *)"\x3c\x0f\x00", 3, 4);
-		if (base) {
-			plt_addr += (int)(size_t) (base - buf);
-		} else {
-			plt_addr += RISCV_PLT_OFFSET; // HARDCODED HACK
+		if (plt_sym_addr == rel->rva) {
+			return plt_addr;
 		}
-		plt_addr += rel->k * 16;
-		return plt_addr;
-	}
-	if (plt_section) {
-		const int sizeOfProcedureLinkageTable = 32;
-		const int sizeOfPltEntry = 16;
-		return plt_section->rva + sizeOfProcedureLinkageTable + (rel->k * sizeOfPltEntry);
+		plt_addr += 8;
 	}
-	eprintf ("Unsupported relocs type %" PFMT64u " for arch %d\n",
-		(ut64)REL_TYPE, bin->ehdr.e_machine);
+
 	return UT64_MAX;
 }
 
-
-static ut64 get_import_addr_mips(ELFOBJ *bin, struct ht_rel_t *rel, RBinElfSection *plt_section) {
-	RBinElfSection *s = get_section_by_name (bin, ".rela.plt");
-	if (s) {
-		ut8 buf[1024];
-		const ut8 *base;
-		ut64 plt_addr = s->rva + s->size;
-		int len = r_buf_read_at (bin->b, s->offset + s->size, buf, sizeof (buf));
-		if (len != sizeof (buf)) {
-			// oops
-		}
-		base = r_mem_mem_aligned (buf, sizeof (buf), (const ut8 *)"\x3c\x0f\x00", 3, 4);
-		if (base) {
-			plt_addr += (int)(size_t) (base - buf);
-		} else {
-			plt_addr += MIPS_PLT_OFFSET + 8; // HARDCODED HACK
-		}
-		plt_addr += rel->k * 16;
-		return plt_addr;
+static ut64 get_import_addr_x86(ELFOBJ *bin, RBinElfReloc *rel) {
```

Comment:
I think you changed too many things... Also, there were some nice comments in the code for x86, like the new layout for endbr64 in plt. Please, try to revert as much as possible to existing code and just do the minimum number of changes to make it work better, without rewriting everything from scratch.